ambry-tools/src/main/java/com.github.ambry/store/DiskReformatter.java

```diff
@@ -0,0 +1,281 @@
+            config.partitionLayoutFilePath);
+    try (ClusterMap clusterMap = clusterAgentsFactory.getClusterMap()) {
+      StoreKeyFactory storeKeyFactory = Utils.getObj(storeConfig.storeKeyFactory, clusterMap);
+      DataNodeId dataNodeId = clusterMap.getDataNodeId(config.datanodeHostname, config.datanodePort);
+      if (dataNodeId == null) {
+        throw new IllegalArgumentException(
+            "Did not find node in clustermap with hostname:port - " + config.datanodeHostname + ":"
+                + config.datanodePort);
+      }
+      DiskReformatter reformatter =
+          new DiskReformatter(dataNodeId, config.fetchSizeInBytes, storeConfig, storeKeyFactory, clusterMap,
+              SystemTime.getInstance());
+      AtomicInteger exitStatus = new AtomicInteger(0);
+      CountDownLatch latch = new CountDownLatch(config.diskMountPaths.length);
+      for (int i = 0; i < config.diskMountPaths.length; i++) {
+        int finalI = i;
+        Runnable runnable = () -> {
+          try {
+            reformatter.reformat(config.diskMountPaths[finalI], new File(config.scratchPaths[finalI]));
+            latch.countDown();
+          } catch (Exception e) {
+            throw new IllegalStateException(e);
+          }
+        };
+        Thread thread = Utils.newThread(config.diskMountPaths[finalI] + "-reformatter", runnable, true);
+        thread.setUncaughtExceptionHandler((t, e) -> {
+          exitStatus.set(1);
+          logger.error("Reformatting {} failed", config.diskMountPaths[finalI], e);
+          latch.countDown();
+        });
+        thread.start();
+      }
+      latch.await();
+      System.exit(exitStatus.get());
+    }
+  }
+
+  /**
+   * @param dataNodeId the {@link DataNodeId} on which {@code diskMountPath} exists.
+   * @param fetchSizeInBytes the size of each fetch from the source store during copy
+   * @param storeConfig the config for the stores
+   * @param storeKeyFactory the {@link StoreKeyFactory} to use.
+   * @param clusterMap the {@link ClusterMap} to use get details of replicas and partitions.
+   * @param time the {@link Time} instance to use.
+   */
+  public DiskReformatter(DataNodeId dataNodeId, long fetchSizeInBytes, StoreConfig storeConfig,
+      StoreKeyFactory storeKeyFactory, ClusterMap clusterMap, Time time) {
+    this.dataNodeId = dataNodeId;
+    this.fetchSizeInBytes = fetchSizeInBytes;
+    this.storeConfig = storeConfig;
+    this.storeKeyFactory = storeKeyFactory;
+    this.clusterMap = clusterMap;
+    this.time = time;
+    consistencyChecker =
+        new ConsistencyCheckerTool(clusterMap, new StoreToolsMetrics(clusterMap.getMetricRegistry()), time);
+  }
+
+  /**
+   * Performs a reformat of the disk.
+   * 1. Copies one partition on the disk to a scratch space
+   * 2. Performs local copies of all other partitions on the disk using {@link StoreCopier} and deletes the source.
```

Comment:
Changed the description a bit. Let me know if it could use a few more changes to be clearer